source/grid/tria.cc

```diff
@@ -9015,6 +9015,35 @@ Triangulation (Triangulation<dim, spacedim> &&tria)
 
   tria.number_cache = internal::Triangulation::NumberCache<dim>();
 }
+
+
+template <int dim, int spacedim>
+Triangulation<dim, spacedim> &
+Triangulation<dim, spacedim>::operator= (Triangulation<dim, spacedim> &&tria)
+{
+  Subscriptor::operator=(std::move(tria));
```

Comment:
Yes that's correct, the move constructor and assignment operator for subscriber call `check_no_subscribers`, which throws if there are any subscribers left.